### PR TITLE
Client SDK: Added functionality to get public key from private key

### PR DESCRIPTION
**Pull request by MartinMinkov**
This PR implements the `publicKeyOfPrivateKey` function which returns a public key corresponding to a private key.

### CODE REVIEW

**Review by mrmr1993 — APPROVED** [review]
LGTM

**Comment by mrmr1993** [comment]
I think this PR should be against compatible, can you change the base branch? (There should be an edit button by the PR title to let you do this)

File: frontend/client_sdk/src/CodaSDK.re [hunk 1]
```diff
@@ -60,6 +60,18 @@ type codaSDK;
...
+
+/**
+  * Return the public key corresponding to a private key
+  *
+  * @param privateKey - The private key used to get the corresponding public key
+  * @returns A public key
+ */
+
+[@genType]
+let publicKeyOfPrivateKey = (. privateKey: privateKey) => publicKeyOfPrivateKey(codaSDK, privateKey);
```
Comment (bkase) [inline frontend/client_sdk/src/CodaSDK.re:71]: As discussed can you change this to `derivePublicKey`?

**Review by bkase — COMMENTED** [review]
And I can help with fixing the git stuff

**Comment by MartinMinkov** [comment]
Due to nuking this PR with some errors with rebase, this PR has been moved too https://github.com/MinaProtocol/mina/pull/7512

**Pull request closed without merging**